front/src/components/User/UsersTable.tsx

```diff
@@ -0,0 +1,204 @@
+
+  const sortSettings = {
+    'email': true,
+    'firstname': true,
+    'lastname': true,
+    'createdAt': true,
+    'updatedAt': true,
+    'deletedAt': true,
+  };
+
+  const getSortedUsers = useCallback(() => {
+    return [...users].sort((a, b) => {
+      switch (sortColumn) {
+      case 'email':
+        return compareStrings(a.email, b.email, sortOrder);
+      case 'firstname':
+        return compareStrings(a.firstname, b.firstname, sortOrder);
+      case 'lastname':
+        return compareStrings(a.lastname, b.lastname, sortOrder);
+      case 'createdAt':
+        return compareDates(a.createdAt, b.createdAt, sortOrder);
+      case 'updatedAt':
+        if (a.updatedAt !== null && b.updatedAt !== null) {
+          return compareDates(a.updatedAt, b.updatedAt, sortOrder);
+        }
+        return 0;
+      case 'deletedAt':
+        if (a.deletedAt !== null && b.deletedAt !== null) {
+          return compareDates(a.deletedAt, b.deletedAt, sortOrder);
+        }
+        if(a.deletedAt !== b.deletedAt) {
+          return a.deletedAt? -1 : 1;
+        }
+        return 0;
+      default:
+        return 0;
+      }
+    });
+  }, [sortColumn, sortOrder, users]);
+
+  const sortedUsers = useMemo(() => getSortedUsers(), [getSortedUsers]);
+
+  function chunk<T>(array: T[], size: number): T[][] {
+    if (!array.length) {
+      return [];
+    }
+    const head = array.slice(0, size);
+    const tail = array.slice(size);
+    return [head, ...chunk(tail, size)];
+  }
+
+  const pages = chunk(
+    Array(sortedUsers.length)
+      .fill(0)
+      .map((_, index) => ({
+        ...sortedUsers[index],
+      })),
+    perPage,
+  );
+
+  const rows = pages[page - 1].map((user) => (
+    <TableUserRow
+      previousUrl={previousUrl}
+      key={user.uid}
+      user={user}
+      submitDeletion={submitDeletion}
+    />
+  ));
+
+  return <>
+    <TableWithFilter
+      headers={[
+        'uid',
+        'email',
+        'firstname',
+        'lastname',
+        'createdAt',
+        'updatedAt',
+        'deletedAt',
+        'isAdmin',
+        'actions',
+      ]}
+      onHeaderClick={handleSort}
+      sortColumn={sortColumn}
+      sortOrder={sortOrder}
+      sortSettings={sortSettings}
+      rows={rows}
+    />
+    <PaginationMantine
+      total={pages.length} value={page} onChange={setPage}
+    />
+  </>;
+}
+
+function TableUserRow({
+  user,
+  previousUrl,
+  submitDeletion,
+}: WithPreviousUrl<{
+  user: User
+  submitDeletion: (uid: string) => Promise<void>
+}>) {
+  return <Table.Tr>
+    <Table.Td>{user.uid}</Table.Td>
+    <Table.Td>{user.email}</Table.Td>
+    <Table.Td>{user.firstname}</Table.Td>
+    <Table.Td>{user.lastname}</Table.Td>
+    <Table.Td>{new Date(user.createdAt).toDateString()}</Table.Td>
+    <Table.Td>{user.updatedAt ? new Date(user.updatedAt).toDateString() : ''}</Table.Td>
+    <Table.Td>{user.deletedAt ? new Date(user.deletedAt).toDateString() : ''}</Table.Td>
```

Comment:
I would have put `.toLocaleDateString()`, I think it looks better in this context (I don't think we need the day of the week here):
```suggestion
    <Table.Td>{new Date(user.createdAt).toLocaleDateString()}</Table.Td>
    <Table.Td>{user.updatedAt ? new Date(user.updatedAt).toLocaleDateString() : ''}</Table.Td>
    <Table.Td>{user.deletedAt ? new Date(user.deletedAt).toLocaleDateString() : ''}</Table.Td>
```
Example:
Show `20/05/2024`, instead of `Mon May 20 2024`

Reply:
Ah oui c'est bien mieux 👍